targets/curie_bsp/setup.py

```diff
@@ -0,0 +1,204 @@
+                continue
+            asm_file = line.endswith('.S')
+            if asm_file:
+                npos = line.find('.S')
+                out_put += 'obj-y += ' + line[0:npos] + '.o\n'
+    for line in dirs:
+        out_put += 'obj-y += ' + line + '/\n'
+
+    file_path = os.path.join(root_dir, 'Kbuild.mk')
+    file_to_be_created = open(file_path, 'w+')
+    file_to_be_created.write(out_put)
+    file_to_be_created.close()
+    for line in dirs:
+        breadth_first_travel(os.path.join(root_dir, line))
+
+
+
+# create Kbuild.mk in jerryscript/
+def create_kbuild_in_jerryscript(jerry_path):
+    breadth_first_travel(os.path.join(jerry_path, 'jerry-core'))
+    breadth_first_travel(os.path.join(jerry_path, 'jerry-libm'))
+    #jerryscript/
+    out_put = '''
+    obj-y += jerry-core/
+    obj-y += jerry-libm/
+    obj-y += targets/
+    subdir-cflags-y += -DCONFIG_MEM_HEAP_AREA_SIZE=10*1024
+    subdir-cflags-y += -DJERRY_NDEBUG
+    subdir-cflags-y += -DJERRY_DISABLE_HEAVY_DEBUG
+    subdir-cflags-y += -DCONFIG_ECMA_COMPACT_PROFILE
+    subdir-cflags-y += -DCONFIG_ECMA_COMPACT_PROFILE_DISABLE_NUMBER_BUILTIN
+    subdir-cflags-y += -DCONFIG_ECMA_COMPACT_PROFILE_DISABLE_NUMBER_BUILTIN
+    subdir-cflags-y += -DCONFIG_ECMA_COMPACT_PROFILE_DISABLE_STRING_BUILTIN
+    subdir-cflags-y += -DCONFIG_ECMA_COMPACT_PROFILE_DISABLE_BOOLEAN_BUILTIN
+    subdir-cflags-y += -DCONFIG_ECMA_COMPACT_PROFILE_DISABLE_ERROR_BUILTINS
+    subdir-cflags-y += -DCONFIG_ECMA_COMPACT_PROFILE_DISABLE_ARRAY_BUILTIN
+    subdir-cflags-y += -DCONFIG_ECMA_COMPACT_PROFILE_DISABLE_MATH_BUILTIN
+    subdir-cflags-y += -DCONFIG_ECMA_COMPACT_PROFILE_DISABLE_JSON_BUILTIN
+    #subdir-cflags-y += -DCONFIG_ECMA_COMPACT_PROFILE_DISABLE_ERROR_BUILTIN
+    subdir-cflags-y += -DCONFIG_ECMA_LCACHE_DISABLE
+    subdir-cflags-y += -DCONFIG_ECMA_PROPERTY_HASHMAP_DISABLE
+    '''
+
+    file_path = os.path.join(jerry_path, 'Kbuild.mk')
+    file_to_be_created = open(file_path, 'w+')
+    file_to_be_created.write(out_put)
+    file_to_be_created.close()
+
+    # jerryscript/targets
+    out_put =  'obj-y += curie_bsp/'
+    file_path = os.path.join(jerry_path, 'targets/Kbuild.mk')
+    file_to_be_created = open(file_path, 'w+')
+    file_to_be_created.write(out_put)
+    file_to_be_created.close()
+
+    # jerryscript/targets/curie_bsp
+    out_put =  'obj-y += source/'
+    file_path = os.path.join(jerry_path, 'targets/curie_bsp/Kbuild.mk')
+    file_to_be_created = open(file_path, 'w+')
+    file_to_be_created.write(out_put)
+    file_to_be_created.close()
+
+    breadth_first_travel(os.path.join(jerry_path, 'targets/curie_bsp/source'))
+
+# create Makefile in wearable_device_sw/projects/curie_bsp_jerry/
+def create_makefile_in_curie(curie_path):
+
+    # Kbuild.mk
+    out_put = '''
+    subdir-cflags-y += -I$(PROJECT_PATH)/quark/include
+    subdir-cflags-y += -I$(PROJECT_PATH)/quark/jerryscript
+    subdir-cflags-y += -I$(PROJECT_PATH)/quark/jerryscript/jerry-core
```

Comment:
IMHO, all the subdirs of jerry-core shouldn't be hardcoded here. It would be inconvenient to maintain this list if anything changes in the core (which can happen without notice as its structure is an internal thing without any promise on stability). A recursive traversal of the dir structure should be used at run time instead.